pulsar-common/src/main/java/org/apache/pulsar/common/util/protobuf/ByteBufCodedOutputStream.java

```diff
@@ -261,7 +260,6 @@ static int makeTag(final int fieldNumber, final int wireType) {
     /** Write an double field, including tag, to the stream. */
     public void writeDouble(final int fieldNumber, double value) throws IOException {
         writeTag(fieldNumber, WireFormat.WIRETYPE_FIXED64);
-        buf.order(ByteOrder.LITTLE_ENDIAN).writeDouble(value);
+        buf.writeLongLE(Double.doubleToLongBits(value));
```

Comment:
don't we have a way to preserve double value instead casting to long?

Reply:
It's not a "casting" to long. The value is anyway encoded as a varint. I remember I've checked when I made the original change. The bit wise output is that same as now, otherwise the test would catch it anyway.